src/main/config.cpp

```diff
@@ -294,12 +294,25 @@ IndexTypeSet &DBConfig::GetIndexTypes() {
 }
 
 void DBConfig::SetDefaultMaxMemory() {
-	auto memory = FileSystem::GetAvailableMemory();
-	if (!memory.IsValid()) {
-		options.maximum_memory = DBConfigOptions().maximum_memory;
+	const char *slurm_mem_per_node = getenv("SLURM_MEM_PER_NODE");
+	const char *slurm_mem_per_cpu = getenv("SLURM_MEM_PER_CPU");
+	const char *slurm_cpus_on_node = getenv("SLURM_CPUS_ON_NODE");
```

Comment:
Can we call `GetSystemMaxThreads` if `SLURM_MEM_PER_CPU` is set instead of duplicating this logic here?